insert_string_tpl.h

```diff
@@ -63,13 +64,16 @@ ZLIB_INTERNAL Pos QUICK_INSERT_STRING(deflate_state *const s, const Pos str) {
 ZLIB_INTERNAL Pos INSERT_STRING(deflate_state *const s, const Pos str, unsigned int count) {
     Pos idx, ret;
     uint8_t *strstart, *strend;
+    uint32_t hash_mask, hash_mask_val;
 
     if (UNLIKELY(count == 0)) {
         return s->prev[str & s->w_mask];
     }
 
     strstart = s->window + str;
     strend = strstart + count - 1; /* last position */
+    hash_mask = s->hash_mask;
+    hash_mask_val = s->hash_mask_val;
 
     for (ret = 0, idx = str; strstart <= strend; idx++, strstart++) {
```

Comment:
Not really related, but I just noticed that ret does not need to be reset on each iteration of the loop.
Initializing it to zero once is enough, since ret only ever gets set when the loop is finished.

It made no difference in my benchmarks (benched as 0.005% faster, but that is well within the margin of variance per benchmarking run), so gcc is likely already smart enough to move it outside of the loop, making this more of a formality. I'd be fine is this change was added as a separate commit in this PR, otherwise I'll wait until this PR is in to avoid merge conflicts.

Reply:
I kind of don't think this PR should be merged. But just kept open while we investigate. The `ret` change should probably be its own PR since I don't ever see this one getting in. I have come to embrace merge conflicts anyways, in the case it does.. :-)

Reply:
I am in the process of doing a PR for insert_string_tpl.h that includes that change and a couple more that I noticed when rewriting the code for my SIMD experiments.